src/cli/backup/groups_test.go

```diff
@@ -128,108 +128,64 @@ func (suite *GroupsUnitSuite) TestValidateGroupsBackupCreateFlags() {
 func (suite *GroupsUnitSuite) TestBackupCreateFlags() {
 	t := suite.T()
 
-	cmd := &cobra.Command{Use: createCommand}
-
-	// persistent flags not added by addCommands
-	flags.AddRunModeFlag(cmd, true)
-
-	c := addGroupsCommands(cmd)
-	require.NotNil(t, c)
-
-	// non-persistent flags not added by addCommands
-	flags.AddAllProviderFlags(c)
-	flags.AddAllStorageFlags(c)
-
-	flagsTD.WithFlags(
-		cmd,
-		groupsServiceCommand,
-		[]string{
-			"--" + flags.RunModeFN, flags.RunModeFlagTest,
-			"--" + flags.BackupFN, flagsTD.BackupInput,
+	cmd := cliTD.SetUpCmdHasFlags(
+		t,
+		&cobra.Command{Use: createCommand},
+		addGroupsCommands,
+		[]cliTD.UseCobraCommandFn{
+			flags.AddAllProviderFlags,
+			flags.AddAllStorageFlags,
 		},
-		flagsTD.PreparedProviderFlags(),
-		flagsTD.PreparedStorageFlags())
-
-	// Test arg parsing for few args
-	args := []string{
-		groupsServiceCommand,
-		"--" + flags.RunModeFN, flags.RunModeFlagTest,
-
-		"--" + flags.GroupFN, flagsTD.FlgInputs(flagsTD.GroupsInput),
-		"--" + flags.CategoryDataFN, flagsTD.FlgInputs(flagsTD.GroupsCategoryDataInput),
-
-		"--" + flags.FetchParallelismFN, flagsTD.FetchParallelism,
-
-		// bool flags
-		"--" + flags.FailFastFN,
-		"--" + flags.DisableIncrementalsFN,
-		"--" + flags.ForceItemDataDownloadFN,
-		"--" + flags.DisableDeltaFN,
-	}
-
-	args = append(args, flagsTD.PreparedProviderFlags()...)
-	args = append(args, flagsTD.PreparedStorageFlags()...)
-
-	cmd.SetArgs(args)
-
-	cmd.SetOut(new(bytes.Buffer)) // drop output
-	cmd.SetErr(new(bytes.Buffer)) // drop output
-
-	err := cmd.Execute()
-	assert.NoError(t, err, clues.ToCore(err))
+		flagsTD.WithFlags(
+			groupsServiceCommand,
+			[]string{
+				"--" + flags.RunModeFN, flags.RunModeFlagTest,
+				"--" + flags.GroupFN, flagsTD.FlgInputs(flagsTD.GroupsInput),
+				"--" + flags.CategoryDataFN, flagsTD.FlgInputs(flagsTD.GroupsCategoryDataInput),
+				"--" + flags.FetchParallelismFN, flagsTD.FetchParallelism,
+				"--" + flags.FailFastFN,
+				"--" + flags.DisableIncrementalsFN,
+				"--" + flags.ForceItemDataDownloadFN,
+				"--" + flags.DisableDeltaFN,
```

Comment:
Should we move all the flags that are not specific to a single service for each group (backup, restore, list, details) into a package level variable, ie things like fail fast, disable incremental, disable delta etc.

Reply:
Two notes: support and scope.

Per support: not all of these flags are equally supported.  Our application of features like ForceDataDownload, concurrency and paging controls, and etc are sometimes half-implemented, sometimes service-specific.  That makes it difficult to cleanly and easily aggregate all these flags as expected.  Here's an issue to track that standardization, we can follow up on it from there: https://github.com/alcionai/corso/issues/4400

Per scope: this request is out of scope for this PR.  I've made a tech-debt issue to track the movement.  Should be open to contribution when anyone has time to do so: https://github.com/alcionai/corso/issues/4399.